manim/mobject/svg/text_mobject.py

```diff
@@ -1056,11 +1037,441 @@ def text2svg(self):
+        pangocairocffi.update_layout(context, layout)
+        if disable_liga:
+            layout.set_markup(
+                f"<span font_features='liga=0,dlig=0,clig=0,hlig=0'>{self.text}</span>"
+            )
+        else:
+            layout.set_markup(self.text)
+        logger.debug(f"Setting Text {self.text}")
+        pangocairocffi.show_layout(context, layout)
+        surface.finish()
+        return file_name
+
+    def _count_real_chars(self, s):
+        """Internally used function.
+        Counts characters that will be displayed.
+        This is needed for partial coloring or gradients, because space
+        counts to the text's `len`, but has no corresponding letter"""
+        count = 0
+        level = 0
+        # temporarily replace HTML entities by single char
+        s = re.sub("&[^;]+;", "x", s)
+        for c in s:
+            if c == "<":
+                level += 1
+            if c == ">" and level > 0:
+                level -= 1
+            elif c != " " and c != "\t" and level == 0:
+                count += 1
+        return count
+
+    def extract_gradient_tags(self):
+        """Internally used function.
+        Used to determine what parts (if any) of the string should be formatted with a gradient.
+        Removes the ``<gradient>`` tag, as it is not part of Pango's markup and would cause an error."""
+        tags = re.finditer(
+            '<gradient\s+from="([^"]+)"\s+to="([^"]+)"(\s+offset="([^"]+)")?>(.+?)</gradient>',
+            self.original_text,
+            re.S,
+        )
+        gradientmap = []
+        for tag in tags:
+            start = self._count_real_chars(self.original_text[: tag.start(0)])
+            end = start + self._count_real_chars(tag.group(5))
+            offsets = tag.group(4).split(",") if tag.group(4) else [0]
+            start_offset = int(offsets[0]) if offsets[0] else 0
+            end_offset = int(offsets[1]) if len(offsets) == 2 and offsets[1] else 0
+
+            gradientmap.append(
+                {
+                    "start": start,
+                    "end": end,
+                    "from": tag.group(1),
+                    "to": tag.group(2),
+                    "start_offset": start_offset,
+                    "end_offset": end_offset,
+                }
+            )
+        self.text = re.sub("<gradient[^>]+>(.+?)</gradient>", r"\1", self.text, 0, re.S)
+        return gradientmap
+
+    def _parse_color(self, col):
+        """Internally used function.
+        Parse color given in ``<color>`` or ``<gradient>`` tags."""
+        if re.match("#[0-9a-f]{6}", col):
+            return col
+        else:
+            return Colors[col.lower()].value
+
+    def extract_color_tags(self):
+        """Internally used function.
+        Used to determine what parts (if any) of the string should be formatted with a custom color.
+        Removes the ``<color>`` tag, as it is not part of Pango's markup and would cause an error."""
```

Comment:
```suggestion
        """Used to determine which parts (if any) of the string should be formatted
        with a custom color.

        Removes the ``<color>`` tag, as it is not part of Pango's markup and would cause an error.
        """
```